.github/copilot-instructions.md

```diff
@@ -0,0 +1,188 @@
+
+### Example Skeleton (Button)
+```tsx
+import * as React from 'react';
+import { cn } from '../../utils/cn';
+
+export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  variant?: 'default' | 'destructive' | 'outline' | 'secondary' | 'ghost' | 'link';
+  size?: 'default' | 'sm' | 'lg' | 'icon';
+  asChild?: boolean;
+}
+
+const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
+  ({ className, variant = 'default', size = 'default', asChild = false, ...props }, ref) => {
+    const Comp = asChild ? Slot : 'button';
+    
+    return (
+      <Comp
+        className={cn(buttonVariants({ variant, size }), className)}
+        ref={ref}
+        {...props}
+      />
+    );
+  }
+);
+Button.displayName = 'Button';
+
+export { Button };
+```
+
+## Storybook Conventions
+- One `.stories.tsx` per component folder.
+- Showcase: default usage, variants, sizes, interactive state (focus, loading, disabled).
+- Use `play` function (if Storybook Interaction Testing is in use) for basic a11y or keyboard scenarios.
+- Provide `args` with sensible defaults; expose `variant`, `size`, `disabled`, etc.
+- Avoid business data; use placeholder labels.
+
+## Testing Guidelines
+(Adjust to existing test runner—if not present, default to a minimal React Testing Library + Vitest/Jest approach.)
+- Prefer interaction tests: focus management, keyboard activation, ARIA attributes.
+- Snapshot only for stable markup (rare).
+- Test variants produce expected class combinations.
+- Example assertion: `expect(screen.getByRole('button', { name: /submit/i })).toHaveAttribute('data-variant','primary')`.
+
+## Tailwind Usage
+- Keep class lists ordered logically: layout → spacing → typography → color → interaction → state.
+- Abstract repeated class groups into utilities or component variants rather than duplicating.
+- Use responsive + dark variants only when necessary; avoid premature complexity.
+
+## Accessibility Checklist (Copilot should internalize)
+- Every interactive element has an accessible name.
+- Custom components that render non-semantic elements must add appropriate role.
+- Manage focus on open/close for overlays.
+- Provide `aria-expanded`, `aria-selected`, `aria-controls` when building composite widgets.
+- Ensure color contrast (WCAG AA) for foreground/background.
+
+## Commit Conventions
+Use Conventional Commits to aid changelog & semantic versioning:
+- feat: new component or significant enhancement
+- fix: bug fix
+- docs: documentation only
+- style: formatting (no logic)
+- refactor: internal restructuring
+- test: adding or improving tests
+- chore: tooling / meta
+Examples:
+- feat(button): add loading state
+- fix(modal): restore focus to trigger on close
+
+## Versioning & Release (Assumed)
+- Semantic Versioning (MAJOR.MINOR.PATCH).
+- New components / backwards-compatible props additions: MINOR.
+- Bug fixes / a11y improvements: PATCH.
+- Breaking prop / removal: MAJOR (list breaking changes clearly).
```

Comment:
add that releases are trigger manually via workflow dispatch where the version is picked.
git tags are not used